packages/manager/apps/pci-savings-plan/src/components/Dashboard/Filters/Filters.tsx

```diff
@@ -55,22 +58,85 @@ const SelectWithLabel = <T extends string>({
   );
 };
 
-const Filters = () => {
-  const [resource, setResource] = useState<ResourceType>(ResourceType.instance);
-  const [flavor, setFlavor] = useState<string>('');
-  const { t } = useTranslation(['dashboard']);
+interface FiltersProps {
+  defaultFilter: SavingsPlanService;
+  savingsPlan: SavingsPlanService[];
+  locale: string;
+}
 
-  const { data: technicalList = [] } = useTechnicalInfo({
-    productCode: InstanceTechnicalName.b3,
-  });
+const Filters = ({ defaultFilter, savingsPlan, locale }: FiltersProps) => {
+  const lastTwelveMonths = useMemo(() => getLastTwelveMonths(locale), [locale]);
 
-  const instancesInformation = useMemo(
-    () => getInstancesInformation(technicalList, t),
-    [technicalList, t],
-  );
-  const activeInformations = instancesInformation.filter(
-    (instance) => instance.category === resource,
+  const [resource, setResource] = useState<ResourceType>(ResourceType.instance);
+  const [flavor, setFlavor] = useState<InstanceTechnicalName>(
+    InstanceTechnicalName.b3,
   );
+  const [model, setModel] = useState<string>(defaultFilter.flavor);
+  const [period, setPeriod] = useState<string>(lastTwelveMonths[0]);
+
+  const { t } = useTranslation(['dashboard', 'listing', 'create']);
+
+  const uniqueFlavors = [
+    ...new Set(savingsPlan.map((plan) => plan.flavor.toLowerCase())),
+  ];
+
+  const instancesInformation = useMemo(() => getInstancesInformation(t), [t]);
+
+  const availableResourcesOptions = useMemo(() => {
+    const hasInstance = savingsPlan.some((plan) =>
+      isInstanceFlavor(plan.flavor),
+    );
+    const hasRancher = savingsPlan.some((plan) => plan.flavor === 'Rancher');
```

Comment:
I would recommend putting these variables inside useMemo above to avoid re-renders